packages-exp/auth-exp/test/integration/flows/idp.local.test.ts

```diff
@@ -0,0 +1,234 @@
+
+    await auth.signOut();
+
+    // Sign in again
+    await signInWithCredential(auth, credential);
+    expect(auth.currentUser!.email).to.eq(newEmail);
+  });
+
+  it('allows you to set a password', async () => {
+    const credential = GoogleAuthProvider.credential(oauthIdToken);
+    const { user } = await signInWithCredential(auth, credential);
+
+    expect(user.providerData.length).to.eq(1);
+    expect(user.providerData[0].providerId).to.eq('google.com');
+
+    // Set the password and check provider data
+    await updatePassword(user, 'password');
+    expect(user.providerData.length).to.eq(2);
+    expect(user.providerData.map(p => p.providerId)).to.contain.members([
+      'google.com',
+      'password'
+    ]);
+
+    // Sign out and sign in again
+    await auth.signOut();
+    await signInWithEmailAndPassword(auth, email, 'password');
+    expect(auth.currentUser!.providerData.length).to.eq(2);
+    expect(
+      auth.currentUser!.providerData.map(p => p.providerId)
+    ).to.contain.members(['google.com', 'password']);
+
+    // Update email, then sign out/sign in again
+    const newEmail = randomEmail();
+    await updateEmail(auth.currentUser!, newEmail);
+    await auth.signOut();
+    await signInWithEmailAndPassword(auth, newEmail, 'password');
+    expect(auth.currentUser!.providerData.length).to.eq(2);
+    expect(
+      auth.currentUser!.providerData.map(p => p.providerId)
+    ).to.contain.members(['google.com', 'password']);
+  });
+
+  it('can link with multiple idps', async () => {
+    const googleEmail = randomEmail();
+    const facebookEmail = randomEmail();
+
+    const googleCredential = GoogleAuthProvider.credential(
+      JSON.stringify({
+        sub: googleEmail,
+        email: googleEmail,
+        email_verified: true
+      })
+    );
+
+    const facebookCredential = FacebookAuthProvider.credential(
+      JSON.stringify({
+        sub: facebookEmail,
+        email: facebookEmail
+      })
+    );
+
+    // Link and then test everything
+    const { user } = await signInWithCredential(auth, facebookCredential);
+    await linkWithCredential(user, googleCredential);
+    expect(user.email).to.eq(facebookEmail);
+    expect(user.emailVerified).to.be.false;
+    expect(user.providerData.length).to.eq(2);
+    expect(
+      user.providerData.find(p => p.providerId === 'google.com')!.email
+    ).to.eq(googleEmail);
+    expect(
+      user.providerData.find(p => p.providerId === 'facebook.com')!.email
+    ).to.eq(facebookEmail);
+
+    // Unlink Google and check everything again
+    await unlink(user, ProviderId.GOOGLE);
+    expect(user.email).to.eq(facebookEmail);
+    expect(user.emailVerified).to.be.false;
+    expect(user.providerData.length).to.eq(1);
+    expect(user.providerData[0].email).to.eq(facebookEmail);
+    expect(user.providerData[0].providerId).to.eq('facebook.com');
+  });
+
+  it('IdP account takes over unverified email', async () => {
```

Comment:
related: would you mind adding one more test for two IDPs with verified emails (e.g. Google and GitHub) and then make sure they are automatically linked (without even using `link(...)`)?